### PR TITLE
[FEATURE] Add context.data_sources.all

### PR DESCRIPTION
**Pull request by billdirks**
- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: tests/integration/test_script_runner.py [hunk 1]
```diff
@@ -384,8 +383,7 @@ def _execute_integration_test(  # noqa: C901, PLR0915
         base_dir = pathlib.Path(file_relative_path(__file__, "../../"))
         os.chdir(base_dir)
         # Ensure GX is installed in our environment
-        installed_packages = [pkg.key for pkg in pkg_resources.working_set]
-        if "great-expectations" not in installed_packages:
+        if importlib.util.find_spec("great-expectations") is None:
```
Comment (billdirks) [inline tests/integration/test_script_runner.py]: I think we should probably just die here instead. Requiring installation of GX before running integration tests seems reasonable.

File: great_expectations/data_context/data_context/abstract_data_context.py [hunk 2]
```diff
@@ -1918,15 +1918,13 @@ def variables(self) -> DataContextVariables:
     def progress_bars(self) -> Optional[ProgressBarsConfig]:
         return self.variables.progress_bars
 
-    @property
-    def datasources(self) -> DatasourceDict:
```
Comment (billdirks) [inline great_expectations/data_context/data_context/abstract_data_context.py:1922]: Deleting this property is a core change necessitating the other changes.

File: great_expectations/datasource/fluent/sources.py [hunk 3]
```diff
@@ -593,6 +590,10 @@ def delete(self, name: str) -> None:
         """
         self._data_context.delete_datasource(datasource_name=name)
 
+    @public_api
+    def all(self) -> DatasourceDict:
```
Comment (billdirks) [inline great_expectations/datasource/fluent/sources.py:594]: This adds the `all` method.

File: great_expectations/datasource/fluent/sources.py [hunk 4]
```diff
@@ -542,15 +541,13 @@ def add_or_update_datasource(
             )
 
             # preserve any pre-existing id for usage with cloud
-            id_: uuid.UUID | None = getattr(
-                self._data_context.datasources.get(datasource_name), "id", None
-            )
+            id_: uuid.UUID | None = getattr(self.all().get(datasource_name), "id", None)
```
Comment (Kilo59) [inline great_expectations/datasource/fluent/sources.py:544]: Does this mean that we are now requesting every single datasource anytime we need to update a single field on a single datasource?

Comment (billdirks) [reply great_expectations/datasource/fluent/sources.py:544]: The behavior should be identical. Previously we had on abstract data context:
```
def datasources(self) -> DatasourceDict:
    return self._datasources
```

Now we have on `_SourceFactories`:
```
    def all(self) -> DatasourceDict:
        return self._data_context._datasources
```

The actual requests are managed internal to `context._datasources`.

**Review by cdkini — APPROVED** [review]

**Pull request merged**